### PR TITLE
Handling the case when response body is empty.

### PR DESCRIPTION
**Pull request by mnmallea**
# Summary
- Fixed the issue described in #37.
- Added test cases to consider responses with empty bodies.

### CODE REVIEW

File: spec/wor/test_services/some_service_head_succesful_spec.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            expect { @response }.not_to raise_error
+          end
+
+          it 'logger.info has been called twice' do
+            expect(service.logger.spied_info_counter).to be(2)
+          end
+        end
+
+        context 'when calling with block' do
+          subject(:service) { described_class.new }
```
Comment (nicozar95) [inline spec/wor/test_services/some_service_head_succesful_spec.rb]: You are repeating the subject here and in the above context, move this subject into the parent context `context 'when calling with attempting_to field defined'` of both:
- `context 'when calling without block' do `
- `context 'when calling with block'`

File: wor-requests.gemspec [hunk 2]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'webmock'
   spec.add_development_dependency 'byebug', '~> 9.0'
   spec.add_development_dependency 'rubocop', '~> 0.47.0'
-  spec.add_development_dependency 'bundler', '~> 1.13'
+  spec.add_development_dependency 'bundler'
```
Comment (nicozar95) [inline wor-requests.gemspec]: Why remove the bundler version?

**Comment by blacksam07** [comment]
Change your destination branch  to `development`

File: spec/wor/test_services/some_service.rb [hunk 3]
```diff
@@ -33,7 +35,9 @@ def method_without_log_without_block
   def method_without_log_with_block
     get(
       path: '/mypath'
-    ) { |response| JSON.parse(response.body)['array'].reduce('', :+) }
+    ) do |response|
+      JSON.parse(response.body)['array'].reduce('', :+) if response.body.present?
```
Comment (wleoguty91) [inline spec/wor/test_services/some_service.rb]: This line is equal to the line in method `method_with_log_with_block`... You can refactor this, move to another method and avoid make repetition in your code

**Review by wleoguty91 — CHANGES_REQUESTED** [review]

**Review by holywyvern — APPROVED** [review]

File: spec/wor/test_services/some_service_head_succesful_spec.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          it 'raises no exception' do
+            expect { @response }.not_to raise_error
+          end
+
+          it 'logger.info has been called twice' do
+            expect(service.logger.spied_info_counter).to be(2)
+          end
+        end
+
+        context 'when calling with block' do
```
Comment (wleoguty91) [inline spec/wor/test_services/some_service_head_succesful_spec.rb]: This context is the same as the context above ... This is not right :confused:

File: spec/wor/test_services/some_service_head_succesful_spec.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          it 'raises no exception' do
+            expect { @response }.not_to raise_error
+          end
+
+          it 'logger.info should not have been called' do
+            expect(service.logger.spied_info_counter).to be(0)
+          end
+        end
+
+        context 'when calling with block' do
```
Comment (wleoguty91) [inline spec/wor/test_services/some_service_head_succesful_spec.rb]: Same here

**Review by wleoguty91 — CHANGES_REQUESTED** [review]

**Review by wleoguty91 — APPROVED** [review]

**Review by jhonmunoz — APPROVED** [review]
lgtm 🍪

**Pull request merged**